### PR TITLE
Ignore source files when publishing

### PR DESCRIPTION
**Pull request by gyfchong**
## Why
For unknown reasons, source files are being published together with compiled files, effectively ignoring the "files" set in package.json. For reasons, consumer apps are picking up source files and checking types/linting.

## What
Add an npmignore file to force npm publish to ignore the files.

To test this, `cd` into any package and run `npm pack --dry-run` which will show you what files are being published.

### CODE REVIEW

File: draft-packages/avatar/.npmignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
+**/*.tsx
+**/*.spec.*
+.turbo/*
+docs
+tsconfig.dist.json
```
Comment (HeartSquared) [inline draft-packages/avatar/.npmignore:6]: You should enable your IDE setting that adds the EOF new line 😛

File: draft-packages/badge/.npmignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline draft-packages/badge/.npmignore:1]: This is per package?! Gross :'(

Comment (HeartSquared) [reply draft-packages/badge/.npmignore:1]: Do you need to ignore `.npmignore` too? Or is that just done by default?

File: draft-packages/form/.npmignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline draft-packages/form/.npmignore:1]: Where's the `package.json` update for form :P

File: draft-packages/illustration/.npmignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline draft-packages/illustration/.npmignore:1]: Where's the `package.json` update for illustration :P

File: draft-packages/menu/.npmignore [hunk 5]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline draft-packages/menu/.npmignore:1]: Where's the `package.json` update for menu :P

File: draft-packages/tabs/.npmignore [hunk 6]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline draft-packages/tabs/.npmignore:1]: Where's the `package.json` update for tabs :P

File: draft-packages/tile/.npmignore [hunk 7]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline draft-packages/tile/.npmignore:1]: Where's the `package.json` update for tile :P

File: packages/brand-moment/.npmignore [hunk 8]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline packages/brand-moment/.npmignore:1]: Where's the `package.json` update for brand-moment :P

File: packages/component-library/.npmignore [hunk 9]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline packages/component-library/.npmignore:1]: Where's the `package.json` update for component-library :P

File: packages/deprecated-component-library-helpers/.npmignore [hunk 10]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline packages/deprecated-component-library-helpers/.npmignore:1]: Where's the `package.json` update for this :P

File: packages/design-tokens/.npmignore [hunk 11]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline packages/design-tokens/.npmignore:1]: Where's the `package.json` update for design-tokens :P

File: packages/hosted-assets/.npmignore [hunk 12]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline packages/hosted-assets/.npmignore:1]: Is this needed if `files` holds no negatives?

File: packages/notification/.npmignore [hunk 13]
```diff
@@ -0,0 +1,6 @@
+**/*.ts
```
Comment (HeartSquared) [inline packages/notification/.npmignore:1]: Where's the `package.json` update for notification :P

File: draft-packages/badge/.npmignore (hunk 2 above)
Comment (gyfchong) [reply draft-packages/badge/.npmignore:1]: I think it needs to be committed, otherwise the pipeline won't be able to read it

Comment (HeartSquared) [reply draft-packages/badge/.npmignore:1]: Yeah definitely committed, but does it need to exist in the dist? This file tells it what to include in the distribution right?

Comment (gyfchong) [reply draft-packages/badge/.npmignore:1]: Lol actually, this files definition works fine:
```
    "**/*.js",
    "**/*.map.js",
    "**/*.d.ts"
```

Comment (HeartSquared) [reply draft-packages/badge/.npmignore:1]: True! As long as we haven't lost anything in between :)

**Comment by gyfchong** [comment]
> One thing to note is that if anyone adds any other type of file to the package (eg. images) then we need to update the `package.json`. Unlikely to happen so often, but we certainly need to keep it in mind (perhaps we need to add some documentation?)

I'm going to optimistically say that we don't need to document cause we're moving to KAIO which puts everything necessary into `dist` 😛

**Pull request merged**